security/pkg/pki/util/keycertbundle_test.go

```diff
@@ -554,8 +540,10 @@ func TestTimeBeforeCertExpires(t *testing.T) {
 				return
 			}
 
-			if time != tc.expectedTime {
-				t.Fatalf("expected time %v, got %v", tc.expectedTime, time)
+            // One second toleration because x509 cert times have one second of precision
+            tol := time.Second
+			if (expiryDuration - tc.expectedTime).Abs() > tol  {
```

Comment:
Having more precise TimeBeforeCertExpires means that we have to be more careful with our comparisons since x509 certificates are accurate up to the second.